packages/cuba-react-core/src/app/Security.ts

```diff
@@ -1,39 +1,59 @@
-import {action, IObservableArray, observable, ObservableMap} from 'mobx';
-import {CubaApp, EntityAttrPermissionValue, PermissionInfo, RoleInfo} from '@cuba-platform/rest';
+import {action, computed, IObservableArray, observable, ObservableMap} from 'mobx';
+import {CubaApp, EntityAttrPermissionValue, EffectivePermsInfo, PermissionInfo} from '@cuba-platform/rest';
 import {getAttributePermission} from '@cuba-platform/rest/dist-node/security';
 
 export class Security {
 
   @observable permissions?: IObservableArray<PermissionInfo>;
   @observable attrPermissionCache: ObservableMap<string, EntityAttrPermissionValue> = new ObservableMap();
-  @observable roles?: IObservableArray<RoleInfo>;
+  @observable effectivePermissions?: EffectivePermsInfo;
   permissionsRequestCount = 0;
 
   constructor(private cubaREST: CubaApp) {
   }
 
+  @computed get dataLoaded(): boolean {
+    return !!this.effectivePermissions || !!this.permissions;
+  };
+
   getAttributePermission = (entityName: string, attributeName: string): EntityAttrPermissionValue => {
 
-    // do not check permissions if roles not set (rest api version prev 7.2)
-    if (!this.roles) return 'MODIFY';
+    // do not check permissions if effectivePermissions not set, but old permissions is loaded (rest api version prev 7.2)
+    if (!this.effectivePermissions && this.permissions) return 'MODIFY';
 
     const attrFqn = `${entityName}:${attributeName}`;
 
     let perm = this.attrPermissionCache.get(attrFqn);
     if (perm != null) return perm;
 
-    perm = getAttributePermission(entityName, attributeName, this.permissions, this.roles);
+    perm = getAttributePermission(entityName, attributeName, this.effectivePermissions);
     this.attrPermissionCache.set(attrFqn, perm);
     return perm;
   };
 
   @action loadPermissions() {
     const requestId = ++this.permissionsRequestCount;
-    this.cubaREST.getPermissions()
-      .then(action((perms: PermissionInfo[]) => {
+
+    this.cubaREST.getEffectivePermissions()
+      .then(action((effectivePermsInfo: EffectivePermsInfo) => {
         if (requestId === this.permissionsRequestCount) {
-          this.permissions = observable(perms);
+          this.effectivePermissions = effectivePermsInfo;
+          this.attrPermissionCache.clear();
         }
-      }));
+      }))
+      .catch(reason => {
+        // support rest api version < 7.2
+        if (reason === CubaApp.NOT_SUPPORTED_BY_API_VERSION) {
+          this.cubaREST.getPermissions()
```

Comment:
Not sure we need to load this permissions for old versions as they do not represent the whole security model

Reply:
Removed.